lark/visitors.py

```diff
@@ -194,18 +203,18 @@ class Transformer_NonRecursive(Transformer):
     Useful for huge trees.
     """
 
-    def transform(self, tree):
+    def transform(self, tree: Tree[_Leaf_T]) -> _Return_T:
         # Tree to postfix
         rev_postfix = []
-        q = [tree]
+        q: List[Union[_Leaf_T, Tree[_Leaf_T]]] = [tree]
         while q:
             t = q.pop()
             rev_postfix.append(t)
             if isinstance(t, Tree):
                 q += t.children
 
         # Postfix to tree
-        stack = []
+        stack: List = []
```

Comment:
Why this addition?

Reply:
Yeah, so this is a fun one:
* There isn't enough type information based on the `stack.append()` calls for `mypy` to understand what should be allowed to be in the list. So it asks for the initialization of the variable to be annotated.
* However, the list can contain anything based on the implementations of the user functions ([See comment](https://github.com/plannigan/lark/blob/732143a708eeef9e76da992684213deb5ef507df/lark/visitors.py#L233-L235)). So that means that the type is `List[Any]`.
* But the generic `Any` is redundant and can be removed.

So we are left we are left with an annotation that seems to be redundant to humans, but is there to explicitly tell `mypy` how the list is intended to be used.